element_electrode_localization/coordinate_framework.py

```diff
@@ -56,13 +71,124 @@ class Voxel(dj.Part):
         -> CCF.Voxel
         """
 
+    @classmethod
+    def retrieve_acronym(self, acronym):
+        """ Retrieve the DataJoint translation of the CCF acronym"""
+        return re.sub(r'(?<!^)(?=[A-Z])', '_', acronym).lower()
+
+    @classmethod
+    def voxel_query(self, x=None, y=None, z=None):
+        """Given one or more coordinates, return unique brain regions
+        :param x: x coordinate
+        :param y: y coordinate
+        :param z: z coordinate
+        """
+        if not any(x, y, z):
+            raise ValueError('Must specify at least one dimension')
+        # query = self.Voxel  #  TODO: add utility function name lookup
+        raise NotImplementedError('Coming soon')
+
 
 @schema
 class ParentBrainRegion(dj.Lookup):
-    definition = """ # Hierarchical structure between the brain regions
+    definition = """ # Hierarchical structure between the brain regionss
     -> BrainRegionAnnotation.BrainRegion
     ---
     -> BrainRegionAnnotation.BrainRegion.proj(parent='acronym')
     """
 
 
+# ---- HELPERS ----
+
+
+def load_ccf_annotation(ccf_id, version_name, voxel_resolution,
+                        nrrd_filepath, ontology_csv_filepath):
+    """
+    :param ccf_id: unique id to identify a new CCF dataset to be inserted
+    :param version_name: CCF version
+    :param voxel_resolution: voxel resolution in micron
+    :param nrrd_filepath: path to the .nrrd file for the volume data
+    :param ontology_csv_filepath: path to the .csv file for the brain region ontology
+
+    load_ccf_annotation(
+        ccf_id=0, version_name='ccf_2017', voxel_resolution=10,
+        nrrd_filepath='./data/annotation_10.nrrd',
+        ontology_csv_filepath='./data/query.csv')
+
+    For an example Allen brain atlas for mouse, see:
+    http://download.alleninstitute.org/informatics-archive/current-release/mouse_ccf/annotation/ccf_2017
+
+    For the structure/ontology tree, see:
+    https://community.brain-map.org/t/allen-mouse-ccf-accessing-and-using-related-data-and-tools/359
+    (particularly the ontology file downloadable as CSV)
+    """
+    ccf_key = {'ccf_id': ccf_id}
+    if CCF & ccf_key:
+        print(f'CCF ID {ccf_id} already exists!')
+        return
+
+    nrrd_filepath = pathlib.Path(nrrd_filepath)
+    ontology_csv_filepath = pathlib.Path(ontology_csv_filepath)
+
+    def to_snake_case(s):
+        return re.sub(r'(?<!^)(?=[A-Z])', '_', s).lower()
+
+    ontology = pd.read_csv(ontology_csv_filepath)
```

Comment:
Using Pandas for reading columns from a CSV file is a bit of an overkill. A simpler method from the standard library should be used.